Ex4/Ex4.java

```diff
@@ -0,0 +1,81 @@
+public class Ex4 {
+
+	public static final boolean LAND = true;
+	public static final boolean WATER = false;
+	
+	public static boolean[][] map = {{WATER, LAND, WATER, LAND},
+					 {LAND, LAND, WATER, WATER},
+					 {WATER, WATER, LAND, WATER},
+					 {WATER, WATER, LAND, WATER}};
+	
+	public static void main(String[] args) {
+		System.out.print("Number of islands in the map: ");
+		System.out.println(countIslands(map));
+	}
+	
+	/**
+	 * iterate over each cell of the map.
+	 * when encountering land, mark all of its consecutive land neighbors
+	 * to form one island
+	 * @param map boolean map where true is land and false is water
```

Comment:
I like the fact your code is robust against non-square maps. But it might be worth mentioning this in the javadoc, and explain how non-square arrays are interpreted (as aligned down the 0 column)